src/main/java/io/confluent/connect/elasticsearch/ClientConfigCallbackHandler.java

```diff
@@ -210,4 +308,61 @@ private SSLContext sslContext() {
       }
     }
   }
+
+  private LoginContext login() throws PrivilegedActionException {
+    Configuration conf = new Configuration() {
+      @Override
+      public AppConfigurationEntry[] getAppConfigurationEntry(String name) {
+        return new AppConfigurationEntry[] {
+            new AppConfigurationEntry(
+                Krb5LoginModule.class.getName(),
+                AppConfigurationEntry.LoginModuleControlFlag.REQUIRED,
+                kerberosConfigs()
+            )
+        };
+      }
+    };
+
+    return AccessController.doPrivileged(
+        (PrivilegedExceptionAction<LoginContext>) () -> {
+          Subject subject = new Subject(
+              false,
+              Collections.singleton(new KerberosPrincipal(config.kerberosUserPrincipal())),
+              new HashSet<>(),
+              new HashSet<>()
+          );
+          LoginContext loginContext = new LoginContext(
+              "ElasticsearchSinkConnector",
+              subject,
+              null,
+              conf
+          );
+          loginContext.login();
+          return loginContext;
+        }
+    );
+  }
+
+  private Map<String, Object> kerberosConfigs() {
+    Map<String, Object> configs = new HashMap<>();
+    configs.put("useTicketCache", "false");
+    configs.put("useKeyTab", "true");
+    configs.put("keyTab", config.keytabPath());
+    //Krb5 in GSS API needs to be refreshed so it does not throw the error
+    //Specified version of key is not available
+    configs.put("refreshKrb5Config", "true");
+    configs.put("principal", config.kerberosUserPrincipal());
+    configs.put("storeKey", "false");
+    configs.put("doNotPrompt", "true");
+    configs.put("isInitiator", "true");
+    return configs;
+  }
+
+  private static Oid spnegoOid() {
+    try {
+      return new Oid("1.3.6.1.5.5.2");
```

Comment:
What does this ID represent? We could use a constant with a descriptive name for this value.

Reply:
it identifies the SPNEGO auth mechanis,